memgpt/client/client.py

```diff
@@ -0,0 +1,92 @@
+import os
+from typing import Dict, List, Union
+
+from memgpt.cli.cli import QuickstartChoice
+from memgpt.cli.cli import set_config_with_dict, quickstart as quickstart_func
+from memgpt.config import AgentConfig
+from memgpt.persistence_manager import PersistenceManager
+from memgpt.server.rest_api.interface import QueuingInterface
+from memgpt.server.server import SyncServer
+
+
+class Client(object):
+    def __init__(
+        self,
+        auto_save: bool = False,
+        quickstart: Union[QuickstartChoice, None] = None,
```

Comment:
@cpacker I thought about defaulting this to `memgpt`, but then the act of simply constructing the client would (re)configure the system.  I feel that's something the caller should decide explicitly.

Reply:
I made a modification so that if config doesn't exist, we default to quickstart openai (maybe makes more sense to default to quickstart memgpt but we can change that easily later in a subsequent PR)